core/kazoo_number_manager/src/knm_number.erl

```diff
@@ -195,21 +195,18 @@ create_phone_number(Number) ->
 reserve(Num, Options) ->
     case ?MODULE:get(Num, Options) of
         {'ok', Number} ->
-            States = [?NUMBER_STATE_RESERVED, ?NUMBER_STATE_DISCOVERY, ?NUMBER_STATE_AVAILABLE],
-            State = knm_phone_number:state(phone_number(Number)),
-            case lists:member(State, States) of
-                'true' ->
-                    Routines = [fun knm_number_states:to_reserved/1
-                                ,fun save_number/1
-                                ,fun dry_run_or_number/1
-                               ],
-                    apply_number_routines(Number, Routines);
-                'false' ->
-                    {'error', {'invalid_state_transition', State}}
-            end;
-        _Other -> _Other
+            attempt(fun do_reserve/1, [Number]);
+        {'error', _R}=E -> E
     end.
 
+-spec do_reserve(knm_number()) -> knm_number_return().
+do_reserve(Number) ->
```

Comment:
Because it is done is `knm_number_states`. It's pretty much the same impl as `knm_number:move`